app/models/miq_server.rb

```diff
@@ -551,10 +553,20 @@ def logon_status_details
   # Zone and Role methods
   #
   def self.my_guid
-    @@my_guid_cache ||= begin
-      guid_file = Rails.root.join("GUID")
-      File.write(guid_file, SecureRandom.uuid) unless File.exist?(guid_file)
-      File.read(guid_file).strip
+    @my_guid_mutex ||= Mutex.new
+    @my_guid_mutex.synchronize { @@my_guid_cache ||= load_or_generate_guid }
+  end
+
+  def self.load_or_generate_guid
+    guid = File.read(GUID_FILE).strip if File.exist?(GUID_FILE)
+    return guid if guid.present?
```

Comment:
For the record, the test failures were caused by `my_guid` returning empty/nil and fail the validation on guid being present.  Since the prior code stripped the result of reading the file, it could never return nil but it could return empty string.  Therefore, this check prevents that specific problem from happening.  So, hopefully, this fixes it.